packages/core/upload/admin/src/utils/getFolderParents.js

```diff
@@ -0,0 +1,20 @@
+import flattenTree from '../components/SelectTree/utils/flattenTree';
+
+const getFolderParents = (folders, currentFolderId) => {
+  const parents = [];
+  const flatFolders = flattenTree(folders);
+  const currentFolder = flatFolders.find(folder => folder.value === currentFolderId);
+
+  let { parent } = currentFolder || {};
```

Comment:
I'm just wondering: when would `currentFolder` ever be falsy? And if so, I guess we could just return `parents`, because otherwise the `while` loop would execute once.

Reply:
you are right, it is not needed here - I guess it is because I tested first on CM PR - it means I will have to fix something there, will remove it for now!

Reply:
done [here](https://github.com/strapi/strapi/pull/13831/commits/82740e16e1da5cfc0c894002fd4da1244fe949e0)

Reply:
oh well I just remembered why I did that, I wanted to return an empty array if no results. I can return it sooner to avoid entering the loop as you said!